pkg/tasks/metricsserver.go

```diff
@@ -129,20 +129,15 @@ func (t *MetricsServerTask) create(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
-
-		mcs, err := t.factory.MetricsClientCerts()
+		metricsCerts, err := t.factory.MetricsServerClientCerts()
 		if err != nil {
-			return fmt.Errorf("initializing metrics-client-cert failed: %w", err)
+			return fmt.Errorf("initializing Metrics Server Client Certs secret failed: %w", err)
 		}
 
-		mcs, err = t.client.WaitForSecret(
-			ctx,
-			mcs,
-		)
+		_, err = t.client.WaitForSecret(ctx, metricsCerts)
```

Comment:
I think we should use what `WaitForSecret` returns to get the latest version of the secret, otherwise we can end up with an empty secret.

Reply:
good catch